common/k8s/Cargo.toml

```diff
@@ -22,14 +22,14 @@ tokio = { version = "1", features = ["rt", "rt-multi-thread"] }
 futures = "0.3"
 thiserror = "1.0"
 parking_lot = "0.11"
-kube = { version = "0.52", default-features = false, features = ["rustls-tls"] }
-kube-runtime = { version = "0.52", default-features = false, features = ["rustls-tls"] }
-k8s-openapi = { version = "0.11", default_features = false, features = ["v1_12"] }
+kube = { version = "0.59", default-features = false, features = ["rustls-tls"] }
```

Comment:
Updating to kube >= 0.59 resolves some issues for the agent running in kubernetes clusters that don't expose dns names for their api-servers